### PR TITLE
Store fmf `context` in results for each test

### PR DESCRIPTION
**Pull request by happz**
The actual context is a combination of CLI and plan's context, and it brings additional information on the test configuration that produced a given result.

Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] extend the test coverage
* [x] update the specification
* [x] adjust plugin docstring
* [x] modify the json schema
* [x] mention the version
* [ ] include a release note

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for implementing this. Looks good! Just one naming suggestion.

**Comment by psss** [comment]
Also, what about adding a simple test coverage for the new `results.yaml` key?

File: spec/plans/results.fmf [hunk 1]
```diff
@@ -63,6 +63,17 @@ description: |
        # String, path to /data directory storing possible test artifacts
        data-path: path/to/test/data
 
+       # Mapping, stores the actual fmf context defined for this test.
+       # It's a combination of the context provided via command line
+       # and plan's `context` key.
+       fmf-context:
```
[the remark this answers is not in the document]
Comment (happz) [reply spec/plans/results.fmf]: Renamed in d642621e

File: tmt/schemas/common.yaml [hunk 2]
```diff
@@ -142,6 +166,56 @@ definitions:
         - type: boolean
         - type: number
 
+  # https://tmt.readthedocs.io/en/stable/spec/plans.html#context
+  # https://tmt.readthedocs.io/en/stable/spec/context.html#spec-context
+  fmf_context:
```
Comment (psss) [inline tmt/schemas/common.yaml:171]:
```suggestion
  context:
```

Let's rename this one as well to make it consistent. The keys `distro, component, trigger...` and friends defined here are not an `fmf` thing. They are part of the `tmt` spec.

**Review by psss — CHANGES_REQUESTED** [review]
Just one leftover.

File: spec/plans/results.fmf (hunk 1 above)
[the remark this answers is not in the document]
Comment (psss) [reply spec/plans/results.fmf]: Thanks!

File: tmt/schemas/common.yaml (hunk 2 above)
Comment (happz) [reply tmt/schemas/common.yaml]: Renamed in 418a0507

**Review by psss — APPROVED** [review]

**Comment by psss** [comment]
Squashed and rebased.

**Pull request merged**